### PR TITLE
Track and generate event parameter names

### LINKED ISSUE #1077: Capture event lambda parameter names from the schema
In the schema, writing:
```kotlin
@Property(3) val onChange: (value: String) -> Unit,
```
should produce in the Widget interface:
```kotlin
override fun onChange(onChange: (value: String) -> Unit)
```
and in the Composable function:
```kotlin
onChange: (value: String) -> Unit,
```
Today we only propagate the type. This will require putting both the name and the type into the schema JSON.

### PR DESCRIPTION
**Pull request by JakeWharton**
This is a breaking change to the schema JSON.

Closes #1077

---

- [x] `CHANGELOG.md`'s "Unreleased" section has been updated, if applicable.

### CODE REVIEW

File: redwood-tooling-codegen/src/test/kotlin/app/cash/redwood/tooling/codegen/ComposeGenerationTest.kt [hunk 1]
```diff
@@ -102,8 +101,8 @@ class ComposeGenerationTest {
 
   @Widget(1)
   data class MultipleChildWidget(
-    @Children(1) val top: @Composable () -> Unit,
-    @Children(2) val bottom: @Composable () -> Unit,
+    @Children(1) val top: () -> Unit,
+    @Children(2) val bottom: () -> Unit,
```
Comment (JakeWharton) [inline redwood-tooling-codegen/src/test/kotlin/app/cash/redwood/tooling/codegen/ComposeGenerationTest.kt:105]: Drive-by fix. These did nothing.

File: redwood-tooling-schema/src/main/kotlin/app/cash/redwood/tooling/schema/apiModel.kt [hunk 2]
```diff
@@ -312,7 +312,7 @@ internal data class ApiWidgetEvent(
   constructor(event: ProtocolEvent) : this(
     tag = event.tag,
     name = event.name,
-    params = event.parameterTypes.joinToString(","),
+    params = event.parameters.joinToString(",") { it.type.toString() },
```
Comment (JakeWharton) [inline redwood-tooling-schema/src/main/kotlin/app/cash/redwood/tooling/schema/apiModel.kt:315]: Names aren't in the ABI so we don't track as part of API dump.

File: redwood-tooling-codegen/src/test/kotlin/app/cash/redwood/tooling/codegen/WidgetGenerationTest.kt [hunk 3]
```diff
@@ -90,4 +90,29 @@ class WidgetGenerationTest {
...
+      EventParameterNameWidget::class,
+    ],
+  )
+  interface EventParameterNameSchema
+
+  @Widget(1)
+  data class EventParameterNameWidget(
+    @Property(1) val none: () -> Unit,
+    @Property(2) val one: (s: String) -> Unit,
+    @Property(2) val mixed: (i: Int, Long) -> Unit,
```
Comment (JakeWharton) [inline redwood-tooling-codegen/src/test/kotlin/app/cash/redwood/tooling/codegen/WidgetGenerationTest.kt]: Whoops. Why isn't this failing?!?

**Review by colinrtwhite — APPROVED** [review]

File: redwood-tooling-codegen/src/test/kotlin/app/cash/redwood/tooling/codegen/ComposeGenerationTest.kt [hunk 4]
```diff
@@ -147,4 +146,29 @@ class ComposeGenerationTest {
...
+      EventParameterNameWidget::class,
+    ],
+  )
+  interface EventParameterNameSchema
+
+  @Widget(1)
+  data class EventParameterNameWidget(
+    @Property(1) val none: () -> Unit,
+    @Property(2) val one: (s: String) -> Unit,
+    @Property(3) val mixed: (i: Int, Long) -> Unit,
```
Comment (swankjesse) [inline redwood-tooling-codegen/src/test/kotlin/app/cash/redwood/tooling/codegen/ComposeGenerationTest.kt:161]: oh neat

**Review by swankjesse — APPROVED** [review]
